augur/dates/__init__.py

```diff
@@ -34,6 +34,15 @@ def numeric_date(date):
     >>> numeric_date("1W") == treetime.utils.numeric_date(datetime.date.today() - isodate.parse_duration("P1W"))
     True
     """
+    try:
+        # Put all logic in another function to have the original date string for
+        # any errors that may arise.
+        return _numeric_date(date)
+    except InvalidDateMessage as error:
+        raise InvalidDate(date, str(error))
```

Comment:
I agree it's not ideal, and future changes was the main reason. Notably ebcb56a3423286a03d628ed532a1fdb69dd230db, since `AmbiguousDate` is initialized with the modified date (e.g. `2018 →  2018-XX-XX`) so errors that come from that class have no visibility to the original unmodified date.

Another option is to parse all date formats in the same class, for example:

```py
def numeric_date(date, fmt, …):
    return Date(date, fmt, …).to_numeric()

class Date:
    def __init__(self, date, fmt, …):
        self.original_date = date
        self.fmt = fmt

    def to_numeric(self):
        # <handle other date formats>

        if self.fmt:
            date = self.custom_format_to_datetime_date()
            return treetime.utils.numeric_date(self.date)

        # <handle other date formats>

    def custom_format_to_datetime_date(self):
        # move AmbiguousDate logic into here and other class methods
        # here we can raise InvalidDate(self.original_date, "custom message")

```

but that seemed like too big of a change.